coders/xcf.c

```diff
@@ -961,8 +1031,8 @@ static MagickBooleanType ReadOneLayer(const ImageInfo *image_info,Image* image,
     }
 
   /* read the hierarchy and layer mask offsets */
-  hierarchy_offset = ReadBlobMSBLong(image);
-  layer_mask_offset = ReadBlobMSBLong(image);
+  hierarchy_offset = ReadBlobPointer(image,inDocInfo);
```

Comment:
Could you change the type of `hierarchy_offset` and `layer_mask_offset` to `MagickOffsetType`? And remove the casts in the lines below?